feat(state): Provide the next proposer in `PrepareProposalRequest`

alesforz opened the pull request.

Commit:
formatted vars in CreateProposalBlock()

Commit:
extract next proposer's address in CreateProposalBlock()

Commit:
extract local last commit in CreateProposalBlock()

Commit:
Added `next_proposer_address` in `PrepareProposalRequest` protobuf

Commit:
regenerated protobuf

Commit:
Added `NextProposerAddress` to `PrepareProposalRequest` in `CreateProposalBlock()`

Commit:
formatted vars in CreateProposalBlock()

melekes reviewed (commented):
lgtm 👍 

can we add some tests and modify `spec/abci/abci++_methods.md`?

Commit:
Updated `PrepareProposal` spec in abci methods doc.

alesforz commented on spec/abci/abci++_methods.md:
My IDE auto-formatter made a lot of changes to this file 😓 But I checked and they did not screw up the layout of the rendered markdown file.
In any case, I've added the new field to the `PrepareProposalRequest` struct.

Commit:
Formatted `TestEmptyPrepareProposal`

Commit:
Added:
- `TestCreateProposalNextProposerAddress` to test that `CreateProposalBlock` adds
  the correct next proposer to the `PrepareProposalRequest`.
- mock `abci.Application` needed by `TestCreateProposalNextProposerAddress`.

melekes commented on proto/cometbft/abci/v1/types.proto:
```diff
@@ -141,6 +141,7 @@ message PrepareProposalRequest {
   bytes next_validators_hash = 7;
   // address of the public key of the validator proposing the block.
   bytes proposer_address = 8;
+  bytes next_proposer_address = 9;
```
please add a comment here too

melekes reviewed (approved):
🚀

Commit:
updated go.mod to address vulnerability in crypto pkg

Commit:
updated github actions go version

Commit:
updated api go.mod to go 1.23.6

Commit:
Revert "updated go.mod to address vulnerability in crypto pkg"

This reverts commit 487f990e29fdac0ebb3785dec6ef3d2c917b4f65.

Updating Go's version should be done in its own PR.

Commit:
Revert "updated github actions go version"

This reverts commit 6f5970fdaa2cc7e72efd4596d53dca43417baa95.

Updating Go's version should be done in its own PR.

Commit:
Revert "updated api go.mod to go 1.23.6"

This reverts commit 40e03a07135ef7ed41e3d09d52f7ab2840ad967d.

Updating Go's version should be done in its own PR.

Commit:
Added doc in `PrepareProposalRequest.next_proposer_address` proto definition.

Commit:
regenerated protobuf

alesforz committed:
Apply suggestions to docs from code review

Co-authored-by: Anton Kaliaev <anton.kalyaev@gmail.com>
Signed-off-by: Alessandro Sforzin <alesforz@gmail.com>

Commit:
regenerate protobuf files

Commit:
Merge remote-tracking branch 'berachain/bera-v1.x' into alesforz/next-proposer-in-proposal

Commit:
Map host port 5433 to postgres docker container 5432.

Fix for the `state/indexer/sink/psql` tests failing with a docker port error.
